### PR TITLE
feat(set-version): reimplement range support

### PR DESCRIPTION
**Pull request by paul-soporan**
**What's the problem this PR addresses?**

Support for ranges in `yarn set version` was removed in #1114 because of GitHub API limitations.

**How did you fix it?**

Reimplemented range support using https://repo.yarnpkg.com/tags.

**Checklist**

- [X] I have read the [Contributing Guide](https://yarnpkg.com/advanced/contributing).

- [X] I have set the packages that need to be released for my changes to be effective.

- [X] I will check that all automated PR checks pass before the PR gets reviewed.

### CODE REVIEW

File: packages/plugin-essentials/sources/commands/set/version.ts [hunk 1]
```diff
@@ -94,10 +99,39 @@ export default class SetVersionCommand extends BaseCommand {
...
+
+      // Ignore prereleases
+      if (candidate.prerelease.length > 0)
+        return false;
+
+      if (!range.test(candidate))
+        return false;
+
+      return true;
+    } catch { }
```
Comment (arcanis) [inline packages/plugin-essentials/sources/commands/set/version.ts:119]: Why do we need a `try/catch`?

File: packages/plugin-essentials/sources/commands/set/version.ts [hunk 2]
```diff
@@ -94,10 +99,39 @@ export default class SetVersionCommand extends BaseCommand {
...
+
+  const data: Tags = await httpUtils.get(`https://repo.yarnpkg.com/tags`, {configuration, jsonResponse: true});
+
+  const candidates = data.tags.filter(version => {
+    try {
+      const candidate = new semverUtils.SemVer(version);
+
+      // Ignore prereleases
+      if (candidate.prerelease.length > 0)
+        return false;
```
Comment (arcanis) [inline packages/plugin-essentials/sources/commands/set/version.ts:113]: Note that when I said "I'd ignore prereleases" I was thinking "no special treatment for prerelease" - basically we could just use `satisfyWithPrereleases`. It's unlikely anyone will use anything else than `^` or similar as range, and in this context they probably want the prereleases if that's all there is.

File: packages/plugin-essentials/sources/commands/set/version.ts (hunk 1 above)
Comment (paul-soporan) [reply packages/plugin-essentials/sources/commands/set/version.ts]: We don't, thanks for noticing! I copied this code from `NpmSemverResolver` where the try/catch is used to ignore invalid semver. In our case, all of our versions are valid semver so we don't need this.

**Comment by paul-soporan** [comment]
The doc updates will be displayed correctly once https://github.com/arcanis/clipanion/pull/92 is merged.

**Pull request merged**